test/dhtest.c

```diff
@@ -195,12 +201,18 @@ int main(int argc, char *argv[])
  err:
     ERR_print_errors_fp(stderr);
 
-    OPENSSL_free(abuf);
-    OPENSSL_free(bbuf);
-    DH_free(b);
-    DH_free(a);
-    BN_GENCB_free(_cb);
-    BIO_free(out);
+    if (abuf)
+        OPENSSL_free(abuf);
+    if (bbuf)
+        OPENSSL_free(bbuf);
+    if (b)
+        DH_free(b);
+    if (a)
+        DH_free(a);
+    if (_cb)
+        BN_GENCB_free(_cb);
+    if (out)
+        BIO_free(out);
```

Comment:
I mistakenly consider that the OPENSSL_free's free_func is registered and can be changed in some specific condition, as you explained, there will have no-ops while the parameter is null, I'll change it back, thank you for ur advice:)